stan/math/rev/core/arena_matrix.hpp

```diff
@@ -49,25 +49,91 @@ class arena_matrix<MatrixType, require_eigen_dense_base_t<MatrixType>>
           ChainableStack::instance_->memalloc_.alloc_array<Scalar>(size),
           size) {}
 
+ private:
+  template <typename T>
+  constexpr auto get_rows(const T& x) {
```

Comment:
These two functions are a bit hard to understand. Could you add doxygen doc to explain what the intent is here? It's not coming through directly through code.

Reply:
The two function definitions are identical. Is that intentional?

Reply:
> The two function definitions are identical. Is that intentional?

One gets the rows in case the conditional is true and the other gets the cols if the conditional is true. I might compress this into one function